### PR TITLE
Fix: not throw nvidia-smi exception

### PR DESCRIPTION
**Pull request by YanjieGao**
for server no gpu or nvidia-smi not work case

- Nvidia-smi error, will only generate job metrics file

- Nvidia-smi live, will generate job & gpu metrics files

### CODE REVIEW

File: prometheus/exporter/gpu_exporter.py [hunk 1]
```diff
@@ -42,17 +42,17 @@ def parseSmiXmlResult(smi, logDir):
...
 
-def genGpuMetricsFromSmi(logDir): 
+def genGpuMetricsFromSmi(logDir):
     try:
         cmd = "nvidia-smi -q -x"
         smi_output = subprocess.check_output([cmd], shell=True)
         return parseSmiXmlResult(smi_output, logDir)
     except subprocess.CalledProcessError as e:
-        raise RuntimeError("command '{}' return with error (code {}): {}".format(e.cmd, e.returncode, e.output))
+        print("command '{}' return with error (code {}): {}".format(e.cmd, e.returncode, e.output))
```
Comment (fanyangCS) [inline prometheus/exporter/gpu_exporter.py]: >print [](start = 8, length = 5)

1. not familiar with Python, do you need to explicitly return null here?
2. can you capture only exception caused by command not found and raise other exception?

Comment (YanjieGao) [reply prometheus/exporter/gpu_exporter.py]: for
1 current method return value will be null when exception trigger. no need to explicitly return null.
2 refine at next commit

File: prometheus/exporter/gpu_exporter.py [hunk 2]
```diff
@@ -42,17 +42,20 @@ def parseSmiXmlResult(smi, logDir):
...
     try:
         cmd = "nvidia-smi -q -x"
         smi_output = subprocess.check_output([cmd], shell=True)
         return parseSmiXmlResult(smi_output, logDir)
     except subprocess.CalledProcessError as e:
-        raise RuntimeError("command '{}' return with error (code {}): {}".format(e.cmd, e.returncode, e.output))
+        if e.returncode == 127:
+            print("command '{}' return with error (code {}): {}".format(e.cmd, e.returncode, e.output))
+        else:
+            raise RuntimeError("command '{}' return with error (code {}): {}".format(e.cmd, e.returncode, e.output))
```
[the remark this answers is not in the document]
Comment (YanjieGao) [reply prometheus/exporter/gpu_exporter.py]: yes will stop. i also suggest and agree use print and continue job exporter other modules work. will refine this

[the remark this answers is not in the document]
Comment (fanyangCS) [reply prometheus/exporter/gpu_exporter.py]: agree we can print other exception here. in the future, we might consider provide a warning for returncode other than 127.

---
In reply to: [182299251](https://github.com/Microsoft/pai/pull/487#discussion_r182299251) [](ancestors = 182299251)

**Pull request merged**